### PR TITLE
Patch NodeStatus conserving NodeConditions

### PR DESCRIPTION
**Pull request by gabrielhartmann**
Resolves Issue #807
Borrows technique from node-problem-detector:
https://github.com/kubernetes/node-problem-detector/blob/a442e71190e7d460c892474336c38a412b332be1/pkg/exporters/k8sexporter/problemclient/problem_client.go#L126-L133

### CODE REVIEW

**Comment by gabrielhartmann** [comment]
I don't believe this change regresses the test introduced here: https://github.com/virtual-kubelet/virtual-kubelet/pull/557/files

File: node/node.go [hunk 1]
```diff
@@ -408,11 +407,12 @@ var emptyGetOptions = metav1.GetOptions{}
 
 // patchNodeStatus patches node status.
 // Copied from github.com/kubernetes/kubernetes/pkg/util/node
```
Comment (sargun) [inline node/node.go]: Can you borrow the latest code from kubernetes upstream? https://github.com/kubernetes/kubernetes/blob/master/pkg/util/node/node.go#L175-L206

Comment (sargun) [reply node/node.go]: Or: https://github.com/kubernetes/kubernetes/blob/master/pkg/util/node/node.go#L256-L268

**Comment by sargun** [comment]
Can you verify node conditions are preserved?

**Comment by gabrielhartmann** [comment]
In my experiments this works to a degree.  `NodeCondition` objects are no longer entirely removed by status updates from VK.  However the result of the patch consistently modify out of band NodeConditions.  Particularly their `Status`.

**Comment by gabrielhartmann** [comment]
If the test added here is run on `master` the patchBytes generated are:
```
{
	"status": {
		"$setElementOrder/conditions": [{
			"type": "Ready"
		}],
		"conditions": [{
			"$patch": "delete",
			"type": "CustomProblem"
		}]
	}
}
```

**Comment by pires** [comment]
I couldn't go through all the code in enough detail to revive my memory of how we implemented the node controller, but IIRC we store the `Node` in the node controller and do not react to node update events, hence the node controller wouldn't know about changes made by others.

**Pull request closed without merging**